### PR TITLE
feat(nestjs/core): run `onApplicationListen` hook after listen server…

### PR DESCRIPTION
**Pull request by underfin**
…. #2795

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: 2795

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

**Comment by underfin** [comment]
Maybe I need some help.

File: packages/core/nest-application.ts [hunk 1]
```diff
@@ -224,6 +224,7 @@ export class NestApplication extends NestApplicationContext
     !this.isInitialized && (await this.init());
 
     this.httpAdapter.listen(port, ...args);
+    this.callListenHook();
```
Comment (kamilmysliwiec) [inline packages/core/nest-application.ts]: `callListenHook` is async

**Comment by underfin** [comment]
Maybe I need some help for the error.

**Comment by bakedChicken** [comment]
@underfin @kamilmysliwiec I would like to help you with this request. What can I do?

**Comment by underfin** [comment]
I will resolve conflicts before merge this pr.

**Comment by kamilmysliwiec** [comment]
Out of curiosity: what's the use case for this PR?

**Comment by underfin** [comment]
Let trace https://github.com/nestjs/nest/issues/2795

**Comment by kamilmysliwiec** [comment]
I think this hook might be somewhat ambiguous. What if one has a hybrid application (e.g., 1 HTTP + 2 microservices)? Another example, what if you have a microservice client (e.g., REDIS) which connection is lazily established (which is the default behavior)?

**Comment by underfin** [comment]
I considered emit event after listening with connections for only http appliction, and we should rethink with how provider an way to represent  the  appliction is available.I'm not ideas.

**Comment by kamilmysliwiec** [comment]
As this would be very ambiguous, I've decided to not merge this PR. Instead, one can manually trigger the appropriate method after calling `app.listen()` (and if there are multiple actions that must be taken, the `@nestjs/event-emitter` package can be used to implement that).

**Pull request closed without merging**